src/openloco/graphics/gfx.cpp

```diff
@@ -861,15 +867,46 @@ namespace openloco::gfx
     }
 
     // 0x004C5C69
+    // input:
+    // regs.ax: left
+    // regs.bx: top
+    // regs.dx: right
+    // regs.bp: bottom
     void set_dirty_blocks(int32_t left, int32_t top, int32_t right, int32_t bottom)
     {
-        registers regs;
-        regs.ax = left;
-        regs.bx = top;
-        regs.dx = right;
-        regs.bp = bottom;
-        call(0x004C5C69, regs);
+        if (left < 0)
```

Comment:
would `left = std::max(0, left);` read better? Same goes for top, right `std::min(_ui_width, right)`, bottom